liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/ExecuteMojoUtil.java

```diff
@@ -0,0 +1,311 @@
+                ));
+        createParams.addAll(libertyCommonParams);
+        createParams.addAll(libertyCommonServerParams);
+    }
+    
+    private static ArrayList<String> deployParams;
+    static {
+        deployParams = new ArrayList<>(Arrays.asList(
+                "appsDirectory", "stripVersion", "deployPackages", "timeout", "looseApplication"
+                // executeMojo can not use alias parameters:
+                // "installAppPackages"
+                ));
+        deployParams.addAll(libertyCommonParams);
+        deployParams.addAll(libertyCommonServerParams);
+    }
+    
+    private static ArrayList<String> installFeatureParams;
+    static {
+        installFeatureParams = new ArrayList<>(Arrays.asList("features"));
+        installFeatureParams.addAll(libertyCommonParams);
+    }
+
+    private static final Map<String, String> libertyAliasMap;
+    static {
+        Map<String, String>tempMap = new HashMap<String, String>();
+        tempMap.put("runtimeArtifact", "assemblyArtifact");
+        tempMap.put("runtimeArchive", "assemblyArchive");
+        tempMap.put("runtimeInstallDirectory", "assemblyInstallDirectory");
+        tempMap.put("configFile", "serverXmlFile");
+        tempMap.put("serverEnv", "serverEnvFile");
+        tempMap.put("installAppPackages", "deployPackages");
+        libertyAliasMap = Collections.unmodifiableMap(tempMap);
+    }
+
+    /**
+     * Given the Plugin get the goal execution configuration.
+     *
+     * @param plugin
+     * @param goal
+     * @return configuration for the plugin execution goal
+     */
+    public static Xpp3Dom getPluginGoalConfig(Plugin plugin, String goal, Log log) {
+        Xpp3Dom config = null;
+        String execId = "default";
+        int numExec = 0;
+
+        List<PluginExecution> executions = plugin.getExecutions();
+        if (executions != null) {
+            for (PluginExecution e : executions) {
+                if (e.getGoals() != null && e.getGoals().contains(goal)) {
+                    if (numExec == 0) {
+                        // execution configuration is already merged with the common plugin
+                        // configuration
+                        config = (Xpp3Dom) e.getConfiguration();
+                        execId = e.getId();
+                    }
+                    numExec++;
+                }
+            }
+            if (config == null) {
+                config = (Xpp3Dom) plugin.getConfiguration();
+            }
+        } else {
+            config = (Xpp3Dom) plugin.getConfiguration();
+        }
+        if (numExec > 1) {
+            log.warn(plugin.getArtifactId() + ":" + goal 
+                    + " goal has multiple execution configrations (default to \"" + execId + "\" execution)");
+        }
+        
+        if (config == null) {
+            config = configuration();
+        } else {
+            config = Xpp3Dom.mergeXpp3Dom(configuration(), config);
```

Comment:
Why does this need to merge with an empty `configuration()`?

Reply:
We need a copy and not a reference so that any subsequent changes to the `config` object (e.g. converting alias parameters or additional changes done by `executeMojo()`) will not be affecting the original configuration object returned by `plugin.getConfiguration()`.

Reply:
It might be helpful to mention this as a comment in the code